### ISSUE TITLE
C++ concepts are not linked to namespaces in XML output

### ISSUE THREAD

**Issue by marzer** [doxygen/doxygen#9622]
**Describe the bug**
C++ concepts are not listed as members of their parent namespace's `<compounddef>` in the XML output.

**Expected behavior**
Given this C++:
```cpp
/// \file
/// \brief Code, yo.

#include <concepts>

/// \brief Test namespace.
namespace test
{

    /// @brief A foo.
    struct foo
    {};

    /// @brief A bar.
    class bar
    {};

    /// @brief Determines if a type has a std::hash implementation.
    template <typename T>
    concept hashable = requires(T a)
    {
        { std::hash<T>{}(a) } -> std::convertible_to<std::size_t>;
    };
}
```

I expected to see the concept `hashable` appear in `namespacetest.xml` as an `<innerconcept>` or similar, but I do not. Instead I see this:

```xml
<?xml version='1.0' encoding='UTF-8'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.9.5" xml:lang="en-US">
  <compounddef id="namespacetest" kind="namespace" language="C++">
    <compoundname>test</compoundname>
    <briefdescription>
<para>Test namespace. </para>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <location file="src/code.h" line="7" column="1"/>
  <innerclass refid="classtest_1_1bar" prot="public">test::bar</innerclass>
    <innerclass refid="structtest_1_1foo" prot="public">test::foo</innerclass>
    </compounddef>
</doxygen>
```

It _does_ appear in `index.xml`, so I suppose the internal mapping is correct and it's just missing the code for emitting concepts when serializing namespaces specifically.

**Version**
1.9.5

**Comment by marzer** [comment]
Oh so this also impacts concepts nested in classes too:

```cpp
	/// @brief A foo.
	struct foo
	{
		/// @brief A bar.
		class bar
		{};

		/// @brief Determines if a type has a std::hash implementation.
		template <typename T>
		concept hashable_nested = requires(T a) {
									  {
										  std::hash<T>{}(a)
										  } -> std::convertible_to<std::size_t>;
								  };
	};
```

`structfoo.xml` will contain an `<innerclass>` element for `bar`, but nothing for the concept.

**Comment by albert-github** [comment]
Regarding the fact: 
>

Is this allowed as I get the error:
```
a.h:34:13: error: concept 'hashable_nested' not in namespace scope
   34 |     concept hashable_nested = requires(T a)
```
and I placed the `struct foo` inside the `namespace test`, when it is allowed, please:
- Can you please attach a, small, self contained **compilable** example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Regarding the namespace problem I've just pushed a proposed patch, pull request #9624

**Comment by marzer** [comment]
> Is this allowed as I get the error

Just checked and you're right, they're only allowed at namespace scope, so I guess that part works fine 😅 

> Can you please attach a, small, self contained compilable example [...]

Sure, will do a bit later today.

**Comment by albert-github** [comment]
No example required as it was only necessary when the `concept` was allowed in a `class`.

**Comment by marzer** [comment]
Alright, no worries.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by marzer** [comment]
Are you referring to 59aa82c1385054dadfb636375e7e45be6b95672c? There doesn't appear to be any regression tests associated with it. That'd be standard practice for a codebase of this size + influence surely?

**Comment by albert-github** [comment]
Yes I'm referring to the mentioned commit.
The number of regression tests for doxygen is indeed a bit small.

**Comment by marzer** [comment]
Alright, well I appreciate the fix :)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
